### ISSUE TITLE
Suggest to convert if to switch

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#13438]
The following pattern is very common but it does need re-evaluation of `expr` (specifically if it's a property access) and is somehow unreadable.

``` cs
if(expr == A || expr == B) { ... }
else if(expr == C) { ... }
else if(expr is D d && cond) { ... }
```

Would be nice to be able to refactor it to a `switch` in one go.

``` cs
switch(expr) {
  case A:
  case B:
    ...
    break;
  case C:
    ...
    break;
  case D d when cond:
    ...
    break;
}
```

Same would apply to chained ternary operators and `match` in the expression context.

**Comment by Pilchie** [comment]
Thanks for filing the issue.  This - along with several other conditional related refactorings (invert if, convert to conditional expression, etc) are definitely things we want to get to.

**Comment by alrz** [comment]
Fixed in #16277 /cc @Pilchie
